File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.py
import sys
num = int(sys.stdin.readline())
a = []

for i in range(num):
    b = (list(map(int,sys.stdin.readline().split())))
    a.append(b)
a.sort(key = lambda x:(x[1],x[0]))

for i in range(num):
    print(a[i][0], a[i][1])

File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int num = Integer.parseInt(br.readLine());
        int[][] arr = new int[num][];
        for(int i = 0 ; i<num ; i++){
            arr[i] = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        }
        Arrays.sort(arr, (a,b) -> {
            if(a[1] == b[1]){
                return Integer.compare(a[0], b[0]);
            }
            return Integer.compare(a[1],b[1]);
        });
        
        for(int[] result : arr){
            System.out.println(result[0] + " " + result[1]);
        }
    }
}
